src/rust-crypto/CrossSigningIdentity.ts

```diff
@@ -108,13 +116,20 @@ export class CrossSigningIdentity {
      *   * Upload the private keys to SSSS, if it is set up
      */
     private async resetCrossSigning(authUploadDeviceSigningKeys?: UIAuthCallback<void>): Promise<void> {
+        // XXX: We must find a way to make this atomic, currently if the user do not remember is account password
+        // or 4S passphrase/key the process will fail in a bad state, with keys rotated but not uploaded or saved in 4S.
         const outgoingRequests: Array<OutgoingRequest> = await this.olmMachine.bootstrapCrossSigning(true);
 
+        // If 4s is configured we need to udpate it.
+        const hasKey = await this.secretStorage.hasKey();
+        if (hasKey) {
+            // Update 4S before uploading cross-signing keys
```

Comment:
Sure, will do.
It's to stay consistent with legacy that will first ask the 4S passphrase (to update 4S) then the account password (to upload keys). In case we have some tests on the app side